lib/oneview-sdk/resource/api500/c7000/volume.rb

```diff
@@ -62,16 +62,19 @@ def delete(flag = :all)
         # @param [OneviewSDK::StoragePool] storage_pool Storage pool.
         def set_storage_pool(storage_pool)
           assure_uri(storage_pool)
-          @data['properties'] ||= {}
-          @data['properties']['storagePool'] = storage_pool['uri']
+          @data['properties']['storagePool'] = storage_pool['uri'] if @data['properties']
         end
 
         # Sets the snapshot pool to the volume
         # @param [OneviewSDK::StoragePool] storage_pool Storage Pool to use for snapshots.
         def set_snapshot_pool(storage_pool)
           assure_uri(storage_pool)
-          @data['properties'] ||= {}
-          @data['properties']['snapshotPool'] = storage_pool['uri']
+          if @data['properties']
```

Comment:
Maybe can be better check if this resource will be used to create a new resource or update the current resource, we can check if the 'uri' attribute is set.
And we should document this behavior.

Reply:
But if the user is trying to create something that already exists or in the case of the 'get_volume_template_uri' method I can assign a storage_pool without the properties attribute? I think it better to do the update treatment as it is done in the create.

Reply:
> But if the user is trying to create something that already exists 

I did not understand the reason for this.

> the case of the 'get_volume_template_uri' method I can assign a storage_pool without the properties attribute?

but the `get_volume_template_uri` method seems getting the value from correct place:
```ruby
 def get_volume_template_uri(template_data)
   storage_pool_uri = self['storagePoolUri'] || self['properties']['storagePool']
   ...
```

Reply:
Ok! Maybe that can solve. ```@data['properties'] ||= {} unless @data['uri']```

Because it does not make much sense for me to check the existence of a volume in this method. If uri was passed, I assume user knows that the resource already exists.